File: code/compatibility/src/main/java/org/betonquest/betonquest/compatibility/nexo/NexoIntegratorFactory.java
package org.betonquest.betonquest.compatibility.nexo;

import org.betonquest.betonquest.compatibility.Integrator;
import org.betonquest.betonquest.compatibility.IntegratorFactory;

public class NexoIntegratorFactory implements IntegratorFactory {

    @Override
    public Integrator getIntegrator() {
        return new NexoIntegrator();
    }
}

File: code/compatibility-kotlin/src/main/kotlin/org.betonquest.betonquest.compatibility/nexo/NexoIntegratorFactory.kt
package org.betonquest.betonquest.compatibility.nexo

import org.betonquest.betonquest.compatibility.Integrator
import org.betonquest.betonquest.compatibility.IntegratorFactory

/**
 * Factory for creating {@link NexoIntegrator} instances.
 */
class NexoIntegratorFactory : IntegratorFactory {

    /**
     * Creates a new instance of the factory.
     */
    override fun getIntegrator(): Integrator = NexoIntegrator()
}
